src/parser.cc

```diff
@@ -664,7 +729,12 @@ bool readDoc(tinyxml2::XMLDocument *_xmlDoc, ElementPtr _sdf,
     if (_convert
         && strcmp(sdfNode->Attribute("version"), SDF::Version().c_str()) != 0)
     {
-      sdfwarn << "Converting a deprecated SDF source[" << _source << "].\n";
+      std::stringstream ss;
+      ss << "Converting a deprecated SDF source[" << _source << "].\n";
+      addRecoverableWarning(
```

Comment:
I'm not sure we want to emit an error for reading an older SDFormat file even when the policy is PEDANTIC. The version of `readDoc` that takes a `SDFPtr` uses `sdfdbg` for this.

Reply:
Switched to sdfdbg

Reply:
Also, with the new commit this should be easy to add a policy for this in the future should it be desired.